.azure-pipelines/template.yml

```diff
@@ -47,6 +47,9 @@ parameters:
 - name: 'publish'
   type: boolean
   default: false
+- name: 'nightly_build'
+  type: number
```

Comment:
Why not a boolean? This would be cleaner I think. You could then conditionally include the env var:
```yaml
            - bash: .azure-pipelines/scripts/test_runner.sh ${{ test_suite }}
              displayName: Run Tests
              env:
                ..
                {{ if parameters.nightly_build }}:
                  SGXLKL_NIGHTLY_BUILD: 1
```